managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -2835,6 +2852,21 @@ public void operationFailed(ManagedLedgerException exception) {
         });
     }
 
+    @VisibleForTesting
+    protected void persistPositionToLedgerWithRetry(LedgerHandle lh, MarkDeleteEntry mdEntry,
+            VoidCallback persistPositionCb, AtomicBoolean isCbCompleted, int timeoutSec) {
+        persistPositionToLedger(lh, mdEntry, persistPositionCb);
+        ledger.getScheduledExecutor().schedule(() -> {
+            if (isCbCompleted.get()) {
+                return;
+            }
+            log.info("[{}] couldn't persist position {} in timeout {} sec for cursor {}, retrying..", ledger.getName(),
+                    mdEntry.newPosition, timeoutSec, name);
+            // keep retrying to recover cursor and enable reads
+            persistPositionToLedgerWithRetry(lh, mdEntry, persistPositionCb, isCbCompleted, timeoutSec);
```

Comment:
if this will retry again i wonder if the ManagedCursorImpl.this lock should be acquired.